openapi/data_repository_service.swagger.yaml

```diff
@@ -105,6 +105,59 @@ paths:
       tags:
         - DataRepositoryService
       x-swagger-router-controller: ga4gh.drs.server
+  '/objects/{object_id}/access/{access_id}':
+    get:
+      summary: Returns a fully resolvable URL that can be used to fetch the actual object bytes.
+      operationId: GetAccessURL
+      security:
```

Comment:
Feels like this PR should either specify security on both methods or neither. I'm okay either way, although I lean slightly towards leaving it out entirely here, since we already know we need better auth support (or at least documentation) in a future PR.

Or we can add exactly the same `security` on both methods. If we do keep it, two questions since I'm not familiar with Swagger `security` syntax:
- does this mean an authToken is **required**? I think we want it to be up to the implementor
- it looks like there's also a way to explicitly say "use OAuth2" -- do we want to support that? (Again optionally.)

Reply:
Ideally all of the API methods would optionally have `security` applied. That way the implementors could choose what to expose publicly and privately for a user. Also please note that all existing methods define 401 error responses, assuming they would all have some sort of auth in place by design.

Current PR suggests requiring authToken only for the `/access` path. Swagger allows making it optional, as well as requiring OAuth2 optionally. Something like this:
```yaml
security:
  - {}
  - authToken: []
  - OAuth2: [] 
```
Which means: "No auth OR authToken OR OAuth2"

If the above `security` is applied to the whole API, individual methods can also override the security rules e.g /service-info method could be:
```yaml
security:
  - {}
```

My suggestion is to update the PR with the following in mind:
- `authToken` is optional in `security` definition
- apply `security` API-wide, ie. all existing GET methods
- leave `OAuth2` for further discussion

That way no existing work would get affected.
As long as `/access` path has a `security` rule (i.e current PR), it'll cover the cases for private/controlled data in the cloud.

Reply:
I like your suggestion ("*apply `security` API-wide, with `authToken` as optional, and `OAuth2` deferred for later*"). Up to you about requiring `service-info` to be public; I kind of like that.

Reply:
Great! On a second thought, making authToken optional can cover both cases for `service-info`; public & private. So I'll just mark it API-wide. Updating PR in a moment.

Reply:
update LGTM; resolving